students/guillaume/session03/slicing_lab.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+def exch_first_last(seq):
+    '''
+    Exchange the first & last item of a sequence
+    '''
+    l = len(seq) - 1
+    seq = (seq[-1:] + seq[1:l] + seq[0:1])
+    return seq
+
+
+def every_other_rem(seq):
+    '''
+    Print a sequence with every other item of the initial sequence
+    '''
+    l = len(seq)
+    seq_ret = seq[0:1]
+    for i in range(2, l, 2):
+        seq_ret = seq_ret + seq[i:i + 1]
+    return seq_ret
+
+
+def first_last_in_bet(seq):
+    '''
+    return a sequence with the first and last 4 items removed,
+    and every other item in between
+    '''
+    return every_other_rem(seq[4:len(seq) - 4])
+
+
+def reverse_slic(seq):
+    '''
+    return a sequence reversed (just with slicing)
+    '''
+    l = len(seq)
+    seq_ret = seq[l - 1:l]
+    for i in range(l - 1, -1, -1):
```

Comment:
this is a for loop -- not "just" slicing!

you did this in the print_grid exercise....